### PR TITLE
Add ReindexingTriggerer to controller maintenance

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs

### CODE REVIEW

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java [hunk 1]
```diff
@@ -81,6 +82,7 @@ public ControllerMaintenance(Controller controller, Metric metric) {
         applicationMetaDataGarbageCollector = new ApplicationMetaDataGarbageCollector(controller, intervals.applicationMetaDataGarbageCollector);
         containerImageExpirer = new ContainerImageExpirer(controller, intervals.containerImageExpirer);
         hostSwitchUpdater = new HostSwitchUpdater(controller, intervals.hostSwitchUpdater);
+        reindexingTriggerer = new ReindexingTriggerer(controller, intervals.reindexingTriggerer);
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java:85]: Should be closed. 🙂

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java:85]: Oooh :)

**Comment by jonmv** [comment]
@mpolden feel free to review as well.

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRepositoryMaintenance.java [hunk 2]
```diff
@@ -59,60 +40,45 @@ public NodeRepositoryMaintenance(NodeRepository nodeRepository, Deployer deploye
...
-        loadBalancerExpirer.ifPresent(NodeRepositoryMaintainer::close);
-        dynamicProvisioningMaintainer.ifPresent(NodeRepositoryMaintainer::close);
-        osUpgradeActivator.close();
-        rebalancer.close();
-        nodeMetricsDbMaintainer.close();
-        autoscalingMaintainer.close();
-        scalingSuggestionsMaintainer.close();
-        switchRebalancer.close();
+        maintainers.forEach(Maintainer::shutdown);
+        maintainers.forEach(Maintainer::close);
```
Comment (mpolden) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/NodeRepositoryMaintenance.java:81]: Better, but why do we need both?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java [hunk 3]
```diff
@@ -28,89 +31,47 @@
...
-        nameServiceDispatcher.close();
-        cloudEventReporter.close();
-        rotationStatusUpdater.close();
-        resourceTagMaintainer.close();
-        systemRoutingPolicyMaintainer.close();
-        applicationMetaDataGarbageCollector.close();
-        containerImageExpirer.close();
-        hostSwitchUpdater.close();
+        maintainers.forEach(Maintainer::shutdown);
+        maintainers.forEach(Maintainer::close);
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java:74]: Better, but why do we need both if close calls shutdown when necessary?

Comment (mpolden) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java:74]: If it's to initiate shutdown on everything at once and then wait, I suggest renaming `close`. `awaitShutdown`?

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/maintenance/ControllerMaintenance.java]: Good idea.

**Review by bjorncs — APPROVED** [review]

**Pull request merged**